notes-20200211.tex

```diff
@@ -0,0 +1,175 @@
+
+\subsection{(Usual) Equality Types}
+
+Next, we introduce equality types $\mathtt{Eq}_A\parens{M; N}$. They are intended to be a direct internalization of judgmental equality by the following equality reflection rule:
+
+\begin{prooftree*}
+	\hypo{\oftype{\ctx}{p}{\mathtt{Eq}_A\parens{M; N}}}
+	\infer1{\eqterm{\ctx}{M}{N}{A}}
+\end{prooftree*}
+
+Note that by the substitutional nature of judgmental equality the converse holds as well:
+
+\begin{prooftree*}
+	\hypo{\eqterm{\ctx}{M}{N}{A}}
+	\infer1{\oftype{\ctx}{\idrefl{M}}{\mathtt{Eq}_A\parens{M; N}}}
+\end{prooftree*}
+
+In this sense, equality types and judgmental equality are equivalent. Formally, the rules of equality types consist of those of identification types and the equality reflection rule.
+
+Note: One implementation of equality types additionally has the following uniqueness rule:
+\begin{prooftree*}
+	\hypo{\oftype{p}{\mathtt{Eq}_A\parens{M; N}}}
+	\infer1[$\mathtt{Eq}$-uniq]{\eqterm{p}{\idrefl}{\mathtt{Eq}_A\parens{M; N}}}
+\end{prooftree*}
+
+\subsection{Reflection-Free Equality Types}
+A variant of equality types is reflection-free equality types \cite{sterling_et_al:LIPIcs:2019:10538}. They are intended to obtain the best parts of identification types and (ordinary) equality types. Reflection-free equality types are obtained from equality types by replacing the equality reflection rule by the following strict principle of uniqueness of identity proofs (UIP):
+
+\begin{prooftree*}
+	\hypo{\oftype{\ctx}{p}{\mathtt{Eq}_A\parens{M; N}}}
+	\hypo{\oftype{\ctx}{q}{\mathtt{Eq}_A\parens{M; N}}}
+	\infer2{\eqterm{\ctx}{p}{q}{\mathtt{Eq}_A\parens{M; N}}}
+\end{prooftree*}
+
+That is, UIP states that there is at most one term of a reflection-free equality type.
+Note that the principle is called \emph{strict} because the equality between terms of the same reflection-free equality type in the conclusion is judgmental. 
+
+\subsection{Identification Types with UIP}
+
+We can also extend identification types with UIP, or alternatively Axiom K. Axiom K is equivalent to UIP in the sense that either of the two is derivable from the other; see section 7.2 of \cite{hott-as:book} for the details. 
+
+\section{The $\infty$-Groupoid Structure of Types}
+
+\begin{theorem}
+Each type $A$ has the structure of an $\infty$-groupoid induced by the identification types constructed from $A$ (recall that a groupoid is a category whose morphisms are all isomorphisms, and an $\infty$-groupoid is an ``infinite-dimensional generalization'' of a groupoid).
+\end{theorem}
+
+This theorem discovers a new connection between type theory and higher category theory. Note, however, that the theorem does not hold if identification types are substituted by equality types or reflection-free equality types introduced below; this is one of the main reasons why identification types are the most popular choice among types on equality.
```

Comment:
The theorem is still true with equality types---it is just not that interesting (because it is "collapsed", lacking non-trivial higher-dimensional structure). It is interesting that identification types already have all the needed structure.